lib/sasl.js

```diff
@@ -10,19 +10,34 @@ var debug = require('debug')('amqp10:sasl'),
 
     Connection = require('./connection');
 
+var saslMechanisms = {
+  PLAIN: 'PLAIN',
+  ANONYMOUS: 'ANONYMOUS',
+  NONE: 'NONE'
+};
+
 /**
- * Currently, only supports SASL-PLAIN
+ * Currently, only supports SASL ANONYMOUS or PLAIN
  *
  * @constructor
  */
-function Sasl() {
+function Sasl(mechanism) {
+  if (mechanism && !u.includes(saslMechanisms, mechanism)) {
+    throw new errors.NotImplementedError(
+      'Only SASL PLAIN and ANONYMOUS are supported.');
+  }
+  this.mechanism = mechanism;
   this.receivedHeader = false;
 }
 
+Sasl.Mechanism = saslMechanisms;
```

Comment:
Although really we're using it like an enum, so its the type of the value, so I think it should be singular.